### PR TITLE
[all] Use new ${env:ENV} style syntax in examples and tests

### LINKED ISSUE #17299: [docs] Use new ${env:ENV} style syntax in examples
### Component(s)

_No response_

### Describe the issue you're reporting

Since open-telemetry/opentelemetry-collector/pull/6276 we support and encourage using `${env:ENV}` style syntax for expanding environment variables in configuration. This syntax is aligned with the expand feature as opposed to the old `$ENV` syntax.

This issue tracks updating documentation and examples in contrib to the new syntax.

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** Use new `${env:ENV}` style syntax in examples and tests

**Link to tracking Issue:** Fixes #17299

### CODE REVIEW

File: exporter/splunkhecexporter/example/splunk.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@ splunk:
       directory: /opt/splunk/etc/apps/search/local
       content:
         metrics:
-          coldPath: $SPLUNK_DB/metrics/colddb
+          coldPath: ${env:SPLUNK_DB}/metrics/colddb
```
Comment (atoulme) [inline exporter/splunkhecexporter/example/splunk.yml:7]: This is a yml file used to configure Splunk Entreprise and is not related to Otel. Please revert those changes.

**Review by atoulme — CHANGES_REQUESTED** [review]
The splunkhec changes are not needed.

File: exporter/splunkhecexporter/example/splunk.yml (hunk 1 above)
Comment (mx-psi) [reply exporter/splunkhecexporter/example/splunk.yml]: Thanks, good catch! Fixed on the latest commit

**Review by atoulme — APPROVED** [review]
Lgtm

File: exporter/sapmexporter/examples/signalfx-k8s.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ data:
       health_check:
       http_forwarder:
         egress:
-          endpoint: "https://api.${SFX_REALM}.signalfx.com"
```
Comment (codeboten) [inline exporter/sapmexporter/examples/signalfx-k8s.yaml:15]: Is this correct? Does the variable get sent to the collector as `${SFX_REALM}` or is it expanded before that?

Comment (mx-psi) [reply exporter/sapmexporter/examples/signalfx-k8s.yaml:15]: @dashpole ping

Comment (dashpole) [reply exporter/sapmexporter/examples/signalfx-k8s.yaml:15]: This shouldn't have anything to do with k8s.  K8s doesn't do any env var expansion within a configmap.

**Review by codeboten — APPROVED** [review]

**Pull request merged**